src/app/dso-shared/dso-edit-metadata/metadata-field-selector/metadata-field-selector.component.html

```diff
@@ -6,13 +6,30 @@
          [formControl]="input"
          (focusin)="query$.next(mdField)"
          (dsClickOutside)="query$.next(null)"
-         (click)="$event.stopPropagation();" />
+         (click)="$event.stopPropagation();"
+         (keyup)="this.selectedValueLoading = false"
+  />
   <div class="invalid-feedback show-feedback" *ngIf="showInvalid">{{ dsoType + '.edit.metadata.metadatafield.invalid' | translate }}</div>
-  <div class="autocomplete dropdown-menu" [ngClass]="{'show': (mdFieldOptions$ | async)?.length > 0}">
-    <div class="dropdown-list">
-      <div *ngFor="let mdFieldOption of (mdFieldOptions$ | async)">
-        <button class="d-block dropdown-item" (click)="select(mdFieldOption)">
-          <span [innerHTML]="mdFieldOption"></span>
+  <div id="scrollable-metadata-field-selector" class="dropdown-menu scrollable-menu" [ngClass]="{'show': (mdFieldOptions$ | async)?.length > 0}">
+    <div  class="dropdown-list">
+      <div
+        infiniteScroll
+        [infiniteScrollDistance]="1"
+        [infiniteScrollThrottle]="0"
+        [infiniteScrollContainer]="'#scrollable-metadata-field-selector'"
+        [fromRoot]="true"
+        (scrolled)="onScrollDown()">
+        <ng-container *ngIf="mdFieldOptions$ | async">
+          <button *ngFor="let listEntry of (mdFieldOptions$ | async)"
+                  class="d-block dropdown-item"
+                  dsHoverClass="ds-hover"
+                  (click)="select(listEntry)" #listEntryElement>
+            <span [innerHTML]="listEntry"></span>
+          </button>
+        </ng-container>
+        <button *ngIf="loading"
+                class="list-group-item list-group-item-action border-0 list-entry">
+          <ds-themed-loading [showMessage]="false"></ds-themed-loading>
```

Comment:
When this is ported to `main`, I believe this `<ds-themed-loading>` tag will need to be `<ds-loading>` because of the changes in 8.0 applied in #2865. 

So, this is an example of why we need a separate PR for `main`.  This current code is correct for 7.x, but will fail in 8.x or `main`.

Reply:
@tdonohue. I merged the main branch with this branch and changed the "ds-loading" import. Thanks for the feedback